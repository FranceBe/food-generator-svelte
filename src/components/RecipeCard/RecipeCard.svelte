<script>
    export let src
    export let title
    export let ingredients
    export let href
</script>
<style>
    .card-container {
        background-color: #FFF;
        border: 1px solid #5F4C44;
        display: flex;
        flex-direction: column;
        width: 80%;
    }

    .title {
        font-weight: 700;
        text-decoration: underline;
        align-self: center;
        padding: 5px;
    }

    .image-container, .place-holder {
        min-width: 200px;
        min-height: 200px;
    }

    .place-holder {
        background-color: #FEFEF1;
        display: flex;
        align-items: center;
        justify-content: center;
    }

    .content {
        display: flex;
        flex-direction: row;
        align-items: center;
        padding: 10px;
    }

    .link-and-ingredients {
        display: flex;
        flex-direction: column;
        padding: 10px;
    }
</style>

<div class="card-container">
    <div class="title">{title}</div>
    <div class="content">
      {#if src}
          <img {src} class="image-container" alt={`${title}`}>
      {:else}
          <div class="place-holder">Pas d&apos;image disponible :(</div>
      {/if}
        <div class="link-and-ingredients">
          {ingredients}
            <a {href}> Lien vers la recette </a>
        </div>
    </div>
</div>
