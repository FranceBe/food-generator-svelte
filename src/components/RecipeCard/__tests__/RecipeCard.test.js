import { render, cleanup } from '@testing-library/svelte'
import RecipeCard from 'components/RecipeCard/RecipeCard.svelte'

describe('RecipeCard', () => {
  afterEach(() => {
    cleanup()
  })
  it('should match snapshot', () => {
    const { container } = render(RecipeCard, {
      props: {
        src: 'image/image.jpg',
        title: 'title of recipe',
        ingredients: 'potatoe, tomatoe',
        href: 'http://my-recipe.recipe',
      }
    })
    expect(container).toMatchSnapshot()
  })
  describe('given there is an src attribute', () => {
    const { container, getAllByText } = render(RecipeCard, {
      props: {
        src: 'image/image.jpg',
        title: 'title of recipe',
        ingredients: 'potatoe, tomatoe',
        href: 'http://my-recipe.recipe',
      }
    })
    it('should find 5 div and 1 img', () => {
      expect(container.getElementsByTagName('div')).toHaveLength(5)
      expect(container.getElementsByTagName('img')).toHaveLength(1)
    })
    it('should render the title', () => {
      expect(getAllByText('title of recipe').length).toBe(1)
    })
  })
})
