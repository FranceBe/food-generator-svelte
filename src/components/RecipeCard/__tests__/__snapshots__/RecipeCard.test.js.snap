// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeCard should match snapshot 1`] = `
<body>
  <div>
    <div
      class="card-container"
    >
      <div
        class="title"
      >
        title of recipe
      </div>
       
      <div
        class="content"
      >
        <img
          alt="title of recipe"
          class="image-container"
          src="image/image.jpg"
        />
         
        <div
          class="link-and-ingredients"
        >
          potatoe, tomatoe
           
          <a
            href="http://my-recipe.recipe"
          >
            Lien vers la recette
          </a>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      class="card-container"
    >
      <div
        class="title"
      >
        title of recipe
      </div>
       
      <div
        class="content"
      >
        <img
          alt="title of recipe"
          class="image-container"
          src="image/image.jpg"
        />
         
        <div
          class="link-and-ingredients"
        >
          potatoe, tomatoe
           
          <a
            href="http://my-recipe.recipe"
          >
            Lien vers la recette
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;
